### PR TITLE
[libc++][concepts] Implements  concept helper `__libcpp_integer`

### PR DESCRIPTION
**Pull request by H-G-Hristov**
...and tests.

### CODE REVIEW

**Comment by H-G-Hristov** [comment]
@mordante @frederick-vs-ja This add `std::__libcpp_integer` concept. 

- Any ideas for an alternative name, e.g. 
  - `std::__libcpp_true_integer`
  - `std::__libcpp_integer_type`
- I have not replaced `__is_safe_integral_cmp` in this patch. I can do it in a follow-up patch.

File: libcxx/test/libcxx/concepts/concepts.arithmetic/arithmetic.compile.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+static_assert(!std::__libcpp_unsigned_integer<short int>);
+static_assert(!std::__libcpp_unsigned_integer<int>);
+static_assert(!std::__libcpp_unsigned_integer<long int>);
+static_assert(!std::__libcpp_unsigned_integer<long long int>);
+static_assert(!std::__libcpp_unsigned_integer<short int>);
+#ifndef _LIBCPP_HAS_NO_INT128
+static_assert(!std::__libcpp_unsigned_integer<__int128_t>);
+#endif
+// Non-integer
+static_assert(!std::__libcpp_unsigned_integer<bool>);
```
Comment (mordante) [inline libcxx/test/libcxx/concepts/concepts.arithmetic/arithmetic.compile.pass.cpp:41]: Please add `char` and `wchar_t` here too, for completeness adding pointer type and enumerate and scoped enumerate would be nice.

File: libcxx/test/libcxx/concepts/concepts.arithmetic/arithmetic.compile.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+#include <concepts>
+
+struct SomeObject {};
+
+// Concept helpers for the internal type traits for the fundamental types.
+
+// template <class _Tp>
+// concept __libcpp_unsigned_integer;
```
Comment (mordante) [inline libcxx/test/libcxx/concepts/concepts.arithmetic/arithmetic.compile.pass.cpp:18]: Typically we have one test per function, type, or concept. Let's split the test in 3 parts for consistency.
Thanks for adding tests for the existing code.

**Review by mordante — COMMENTED** [review]
> @mordante @frederick-vs-ja This adds `std::__libcpp_integer` concept as a prerequisite from this discussion: [#77967 (comment)](https://github.com/llvm/llvm-project/pull/77967#discussion_r1451501514)
> 
>     * Any ideas for an alternative name, e.g.
>       
>       * `std::__libcpp_true_integer`
>       * `std::__libcpp_integer_type`
> 
>     * I have not replaced `__is_safe_integral_cmp` in this patch as suggested [[libc++][numeric] P0543R3: Saturation arithmetic #77967 (comment)](https://github.com/llvm/llvm-project/pull/77967#discussion_r1451099830). I can do it in a follow-up patch.

Thanks. I would keep the name as is. Why do you want to provide different names.

LGTM modulo some small nits.

**Comment by H-G-Hristov** [comment]
```
86\test\std\utilities\format\format.functions\Output\formatted_size.pass.cpp.dir\t.tmp.exe'
# .---command stderr------------
# | Traceback (most recent call last):
# |   File "C:\ws\src\libcxx\utils\run.py", line 72, in <module>
# |     exit(main())
# |   File "C:\ws\src\libcxx\utils\run.py", line 68, in main
# |     return subprocess.call(commandLine, cwd=args.execdir, env=env, shell=False)
# |   File "c:\python39\lib\subprocess.py", line 349, in call
# |     with Popen(*popenargs, **kwargs) as p:
# |   File "c:\python39\lib\subprocess.py", line 951, in __init__
# |     self._execute_child(args, executable, preexec_fn, close_fds,
# |   File "c:\python39\lib\subprocess.py", line 1420, in _execute_child
# |     hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
# | OSError: [WinError 225] Operation did not complete successfully because the file contains a virus or potentially unwanted 
```

**Comment by mordante** [comment]
> ```
> 86\test\std\utilities\format\format.functions\Output\formatted_size.pass.cpp.dir\t.tmp.exe'
> # .---command stderr------------
> # | Traceback (most recent call last):
> # |   File "C:\ws\src\libcxx\utils\run.py", line 72, in <module>
> # |     exit(main())
> # |   File "C:\ws\src\libcxx\utils\run.py", line 68, in main
> # |     return subprocess.call(commandLine, cwd=args.execdir, env=env, shell=False)
> # |   File "c:\python39\lib\subprocess.py", line 349, in call
> # |     with Popen(*popenargs, **kwargs) as p:
> # |   File "c:\python39\lib\subprocess.py", line 951, in __init__
> # |     self._execute_child(args, executable, preexec_fn, close_fds,
> # |   File "c:\python39\lib\subprocess.py", line 1420, in _execute_child
> # |     hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
> # | OSError: [WinError 225] Operation did not complete successfully because the file contains a virus or potentially unwanted 
> ```

I've seen that happen before :-/

**Review by mordante — APPROVED** [review]
LGTM! Thanks! Please ping me when you've update your other patch with these changes.

**Comment by H-G-Hristov** [comment]
> LGTM! Thanks! Please ping me when you've update your other patch with these changes.

Thank you! I'll ping you when I am ready with the other patch. I still need to implement additional tests.

**Pull request merged**